R/fit.R

```diff
@@ -217,13 +228,28 @@ mmrm_control <- function(n_cores = 1L,
   assert_flag(accept_singular)
   assert_flag(drop_visit_levels)
   assert_list(optimizers, names = "unique", types = c("function", "partial"))
+  assert_string(cov, null.ok = TRUE)
   method <- match.arg(method)
+  if (is.null(cov)) {
+    cov <- h_get_cov_default(method)
+  }
+  assert_subset(
+    cov,
+    c("Asymptotic", "Empirical", "Empirical-Jackknife", "Kenward-Roger", "Kenward-Roger-Linear")
+  )
+  if (cov %in% c("Empirical", "Empirical-Jackknife") &&  !identical(method, "Residual")) {
+    stop("Empirical and Empirical-Jackknife only works for Residual degree of freedom currently!")
```

Comment:
```suggestion
    stop("Empirical and Empirical-Jackknife only works for Residual degrees of freedom currently!")
```
etc. best to find/replace across files